surefire-api/src/test/java/org/apache/maven/surefire/api/util/RunOrderCalculatorTest.java

```diff
@@ -59,4 +62,57 @@ public void testOrderTestClasses()
     {
 
     }
+
+    static class DubboLazyConnectTest
+    {
+
+    }
+
+    public void testOrderTestMethods()
+    {
+        RunOrderParameters runOrderParameters = new RunOrderParameters( "testorder" , null );
+        System.setProperty( "test", "MyGoodTest#a2d,MyGoodTest#aBc,MyGoodTest#abc,MyGoodTest#a1b" );
+        DefaultRunOrderCalculator runOrderCalculator = new DefaultRunOrderCalculator( runOrderParameters, 1 );
+        Comparator<String> testOrderRunOrderComparator = runOrderCalculator.comparatorForTestMethods();
+        String[] strArray = { "abc(MyGoodTest)", "a1b(MyGoodTest)", "a2d(MyGoodTest)", "aBc(MyGoodTest)" };
+        List<String> list = Arrays.asList( strArray );
+        list.sort( testOrderRunOrderComparator );
+        String[] strArray2 = { "a2d(MyGoodTest)", "aBc(MyGoodTest)", "abc(MyGoodTest)", "a1b(MyGoodTest)" };
+        List<String> list2 = Arrays.asList( strArray2 );
+        assertEquals( list, list2 );
+    }
+
+    public void testOrderTestClassesAndMethods()
+    {
+        RunOrderParameters runOrderParameters = new RunOrderParameters( "testorder" , null );
+        System.setProperty( "test", "MyBadTest#a2d,MyGoodTest#aBc,MyGoodTest#abc,MyGoodTest#a1b" );
```

Comment:
Why are the test class names MyBadTest and MyGoodTest? What makes the MyBadTest class bad? Not particularly important but you may want to just use TestClass1 and TestClass2 if there aren't specific reasons to call a test class MyBadTest

Reply:
lol